pkg/region/region.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 Outreach Corporation. All Rights Reserved.
+
+// Description: This file contains region specific type definitions
+// and region logic.
+package region
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Name is the name of region
+type Name string
+
+// Region is a logical cloud region
+type region struct {
+	// Name is the name of this region
+	Name Name
+
+	// Cloud is the cloud this region is apart of
+	Cloud CloudName
+
+	// Endpoint is the endpoint to test against
+	Endpoint string
+}
+
+// Duration hits the attached region's endpoint and returns how long it took
+// to do a HEAD request.
+func (r *region) Duration(ctx context.Context) (time.Duration, error) {
+	dur, ok := cache.Get(r.Cloud, r.Name)
+	if ok {
+		return dur, nil
+	}
+
+	startTime := time.Now().UTC()
+	resp, err := http.Head(r.Endpoint) //nolint:gosec // Why: not really variable
+	if err != nil {
+		return 0, err
+	}
+	resp.Body.Close()
+	endTime := time.Now().UTC()
+
+	dur = endTime.Sub(startTime)
```

Comment:
I don't think we need the UTC here (or anywhere for that matter but atleast using UTC makes sense if serialization is involved which isn't the case here).  Also, you can just do `time.Since` : 

```suggestion
  dur := time.Since(startTime)
```